.github/workflows/ci-test.yml

```diff
@@ -0,0 +1,143 @@
+name: ci-test
+
+on:
+  push:
+    branches: [auto, gha-test-*]
+  pull_request:
+    branches: [main, master, release-*, gha-test-*]
+
+jobs:
+  prepare:
```

Comment:
this determines which jobs should run by finding the changes and setting a test-* output to be picked up by dependent jobs.   Right now only file based.   Could also be branch based.    Work around from gha putting the file change attribute only in the "on" section.   So annoying total the wrong place to put that config, should be on the job.   Would make this prepare section unneeded.